### PR TITLE
Add create-meta subcommand

### LINKED ISSUE #143: Multiple entrypoints for CLI
> I think it's a great idea to have a CLI interface. I do think we should have sub commands though. It is likely we would want to be able to create GMF files as well from the command line. `metasynth synthesize` and `metasynth create-metadata` perhaps?

_Originally posted by @qubixes in https://github.com/sodascience/metasynth/pull/142#discussion_r1290709785_

### PR DESCRIPTION
**Pull request by qubixes**
This adds a command to complete the whole pipeline from the command line.

I have added an INI parser to set the specs. An INI file is not ideal, because it has not nesting. However, it is easier to work with than JSON/YAML files, for which you can easily create parsing errors.

fixes #143

### CODE REVIEW

File: metasyn/__main__.py [hunk 1]
```diff
@@ -46,11 +50,72 @@ def main() -> None:
...
+
+def create_metadata():
+    """Program to create metadata from a dataframe."""
+    parser = argparse.ArgumentParser(
+        prog="metasyn create-meta",
+        description="Create a Generative Metadata Format file from a CSV file.",
+    )
+    parser.add_argument(
+        "input",
+        help="input file; a csv file that you want to synthesize later.",
```
Comment (Samuwhale) [inline metasyn/__main__.py:95]:
```suggestion
        help="input file; a CSV file that you want to synthesize later.",
```

Just to keep it consistent with the description of the create_metadata() function

File: metasyn/__main__.py [hunk 2]
```diff
@@ -19,6 +22,7 @@
 Usage: metasyn [subcommand] [options]
 
 Available subcommands:
+    create-meta - generate a GMF/json file.
     synthesize - generate synthetic data from a .json file
```
Comment (Samuwhale) [inline metasyn/__main__.py:26]:
```suggestion
    synthesize - generate synthetic data from a GMF (.json) file 
```

File: metasyn/__main__.py [hunk 3]
```diff
@@ -19,6 +22,7 @@
 Usage: metasyn [subcommand] [options]
 
 Available subcommands:
+    create-meta - generate a GMF/json file.
```
Comment (Samuwhale) [inline metasyn/__main__.py:25]:
```suggestion
    create-meta - generate a GMF (.json) file.
```

A minor tweak to be more in line in how I commonly have worded it on the docs.

**Review by Samuwhale — COMMENTED** [review]
I have not tested it yet, but the code looks good. Just a few very tiny suggestions, but that's it.

File: metasyn/__main__.py [hunk 4]
```diff
@@ -46,11 +50,72 @@ def main() -> None:
...
+                    new_dict[key] = val
+            spec[section[4:]] = new_dict
+    return spec
+
+
+def create_metadata():
+    """Program to create metadata from a dataframe."""
+    parser = argparse.ArgumentParser(
+        prog="metasyn create-meta",
+        description="Create a Generative Metadata Format file from a CSV file.",
```
Comment (Samuwhale) [inline metasyn/__main__.py:91]: Might be nice to explain here that the input CSV has to be a tabular dataset? Not sure if the sentence would be too long then

File: metasyn/__main__.py [hunk 5]
```diff
@@ -46,11 +50,72 @@ def main() -> None:
...
+                except ValueError:
+                    pass
+                if key not in new_dict:
+                    new_dict[key] = val
+            spec[section[4:]] = new_dict
+    return spec
+
+
+def create_metadata():
+    """Program to create metadata from a dataframe."""
```
Comment (Samuwhale) [inline metasyn/__main__.py:88]:
```suggestion
    """Program to create and export metadata from a DataFrame into a GMF file (.json)."""
```

File: metasyn/__main__.py (hunk 4 above)
Comment (qubixes) [reply metasyn/__main__.py:91]: Isn't a CSV file always a tabular dataset though?

Comment (Samuwhale) [reply metasyn/__main__.py:91]: I suppose they often are, you're right. That said, not everyone might be aware of that, in which case it is a lot more user-friendly to ask for an input tabular dataset, despite being redundant (IMO). It doesn't hurt, that's for sure. But if you like it more this way that is cool too.

**Review by vankesteren — APPROVED** [review]
Looks good! Let's merge. In the documentation, we might want to include an example config file. Actually, we can now create CLI documentation yeah?

**Comment by Samuwhale** [comment]
> Looks good! Let's merge. In the documentation, we might want to include an example config file. Actually, we can now create CLI documentation yeah?

I have merged it! There actually already is some CLI documentation, but I'm working on documenting this PR as we speak :)

**Pull request merged**